psydac/feec/derivatives.py

```diff
@@ -193,12 +194,16 @@ def tokronstencil(self):
         # build derivative stencil matrix (don't care for transposition here)
         # hence, use spaceV and spaceW instead of domain, codomain
         periodic_d = self._spaceV.periods[self._diffdir]
+        nc  = self._spaceV.cart.domain_decomposition.ncells[self._diffdir]
         p_d = self._spaceV.pads[self._diffdir]
         n_d = self._spaceV.npts[self._diffdir]
         m_d = self._spaceW.npts[self._diffdir]
 
-        V1_d = StencilVectorSpace([n_d], [p_d], [periodic_d])
-        V2_d = StencilVectorSpace([m_d], [p_d], [periodic_d])
+        domain_1d = DomainDecomposition([nc], [periodic_d])
+        cart1_1d  = CartDecomposition( domain_1d, [n_d], [np.array([0])], [np.array([n_d-1])], [p_d], [1] )
+        cart2_1d  = CartDecomposition( domain_1d, [m_d], [np.array([0])], [np.array([m_d-1])], [p_d], [1] )
```

Comment:
Why do you have to pass Numpy arrays as arguments `global_starts` and `global_ends`?

Reply:
so that we can use NumPy arithmetic operations

Reply:
This sounds quite a bit restrictive.

It is sufficient to add `self._global_starts = np.asarray(global_starts)` to the constructor...

Reply:
done in [9ed13cb](https://github.com/pyccel/psydac/pull/232/commits/9ed13cb7ef3cc0b2cf5ef6284c62b95ffd2874f1)